packages/KDropdownMenu/KDropdownItem.vue

```diff
@@ -0,0 +1,165 @@
+<template>
+  <li
+    :class="{
+      'has-divider': type !== 'link' && hasDivider,
+      'disabled': type === 'default' && disabled,
+      'danger': isDangerous,
+      'k-dropdown-selected-option': selected
+    }"
+    class="k-dropdown-item w-100"
+  >
+    <router-link
+      v-if="type === 'link' && to"
+      :data-testid="label"
+      :class="{ 'disabled': disabled, 'has-divider': hasDivider }"
+      :to="!disabled ? to : $router.currentRoute.path"
+      class="k-dropdown-item-trigger"
+    >
+      <slot>{{ label }}</slot>
+    </router-link>
+    <KButton
+      v-else-if="type === 'button'"
+      :disabled="disabled"
+      :class="classList"
+      class="k-dropdown-item-trigger"
+      v-on="listeners"
+    >
+      <slot>{{ label }}</slot>
+    </KButton>
+    <div
+      v-else
+      class="k-dropdown-item-trigger"
+    >
+      <slot>{{ label }}</slot>
+    </div>
+  </li>
+</template>
+
+<script>
+import KButton from '@kongponents/kbutton/KButton.vue'
+
+export default {
+  name: 'KDropdownItem',
+  components: { KButton },
+  props: {
+    item: {
+      type: Object,
+      default: null,
+      // Items must have a label
+      validator: (item) => item.hasOwnProperty('label')
+    },
+    /**
+     * Use this prop to add a divider above the item
+     */
+    hasDivider: {
+      type: Boolean,
+      default: false
+    },
+    isDangerous: {
+      type: Boolean,
+      default: false
+    },
+    disabled: {
+      type: Boolean,
+      default: false
+    },
+    selected: {
+      type: Boolean,
+      default: false
+    },
+    selectionMenuChild: {
+      type: Boolean,
+      default: false
+    }
```

Comment:
question: I'm not sure I understand when this prob would be used or set to true/false. Can you document it?

Reply:
It's documented:

![image](https://user-images.githubusercontent.com/67973710/182448238-7e26d433-4871-4b3e-829b-52e0e6b69d82.png)

![image](https://user-images.githubusercontent.com/67973710/182448432-cc4b5625-f912-4215-a8e2-56f59544f34e.png)

I will also add a note about it in the `apearance: 'selectionMenu'` section.

Reply:
Reworked this a little bit. I'm not sure whether this is actually useful or not anymore. You can track which element is selected via the standard click events and the prop isn't being used for anything else. I included it originally because `Dropdown` had it I think, but it is really only useful to emit `changed` at the `KDropdownMenu` level rather than at the `KDropdownItem` level but I can't do that because they may be completely slotting the `items` content so there isn't a element that I can guarantee will be there for me to attach the click listener to I don't think.

Can you think of a better way to do this?

Reply:
Ok, refactored again. Now we are using a `handleSelection` method from the `items` slot + `@change` event. This approach seems a lot cleaner. What do you think?

![image](https://user-images.githubusercontent.com/67973710/182683863-1b91b333-d397-4f31-bb61-d269c4bc0c1b.png)